packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -240,4 +251,109 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
       LIMIT 1;
     `);
   }
+
+  async getUsersWhoHaveMadeDialogues(): Promise<DbUser[]> {
+    return this.getRawDb().any(`
+      WITH all_dialogue_authors AS
+        (SELECT (UNNESTED->>'userId') AS _id
+            FROM "Posts" p, UNNEST("coauthorStatuses") unnested
+            WHERE p."collabEditorDialogue" IS TRUE 
+            AND p."draft" IS FALSE
+        UNION
+        SELECT p."userId" as _id
+            FROM "Posts" p
+            WHERE p."collabEditorDialogue" IS TRUE
+            AND p."draft" IS FALSE
+        )
+      SELECT u.*
+      FROM "Users" u
+      INNER JOIN all_dialogue_authors ON all_dialogue_authors._id = u._id
+    `)
+  }
+
+  async getUsersWhoHaveOptedInToDialogueFacilitation(): Promise<DbUser[]> {
+    return this.getRawDb().any(`
+        SELECT *
+        FROM "Users" u
+        WHERE u."optedInToDialogueFacilitation" IS TRUE
+    `)
+  }  
+
+  async getUsersTopUpvotedUsers(userId:string): Promise<UpvotedUser[]> {
+
+    const user = await getUser(userId)
```

Comment:
My guess is that you can just pass in `currentUser` from the place where this is called instead of fetching the user again.